libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2073,6 +2074,22 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 					errorMsg += " Use \"address(" + varName + ")." + memberName + "\" to access this address member.";
 					break;
 				}
+		}
+		if (exprType->category() == Type::Category::Address)
+		{
+			// Trigger error when using send or transfer with a non-payable fallback function.
+			auto addressType = dynamic_cast<AddressType const*>(exprType.get());
```

Comment:
@axic does not particularly like it, but if you use `dynamic_cast<AddressType const&>`, you will get an exception if the type does not match, so you do not need the assert below. The exception is not as nice as the failing assertion below but well...

Reply:
Isn't the assert needed now that it's a pointer?

Reply:
@leonardoalt The condition above in the ``else if`` already asserts that ``exprType`` is an ``AddressType`` - but since that's the case, I'd say it's shorter to just cast in the condition already, as I suggested above.

Reply:
Ah, I misread the suggestion and didn't see the reference part. Will update

Reply:
I used @ekpyron's suggestion now..